File: Python/beginners/traveling.py
n = int(input())
route0 = [[0, 0, 0]]
routeN = list(list(map(int, input().split())) for _ in range(n))
route = route0+routeN

for i in range(len(route)-1):
    current = route[i]
    next = route[i+1]
    dt = next[0] - current[0]
    dist = abs(next[1] - current[1]) + abs(next[2] - current[2])
    if(dist <= dt):
        if not (dist % 2 == dt % 2):
            print("No")
            exit()
    else:
        print("No")
        exit()
print("Yes")

File: C_C++/ABC/086/Traveling.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int N;
    cin >> N;
    vector<int> T(N+1), X(N+1), Y(N+1);
    T[0] = 0, X[0] = 0, Y[0] = 0;
    for(int i = 1; i < N+1; i++) cin >> T[i] >> X[i] >> Y[i];

    int dt, dist;
    bool flag = true;
    for(int i = 0; i < N; i++){
        dt = T[i+1] - T[i];
        dist = abs(X[i+1] - X[i]) + abs(Y[i+1] - Y[i]);
        if(dt < dist){
            cout << "No" << endl;
            exit(0);
        }
        if(dt % 2 != dist % 2){
            cout << "No" << endl;
            exit(0);
        }
    }
    cout << "Yes" << endl;
}